EIPS/eip-7007.md

```diff
@@ -0,0 +1,160 @@
+---
+eip: 7007
+title: zkML AIGC-NFTs
+description: An ERC-721 extension interface for zkML based AIGC-NFTs.
+author: Cathie So (@socathie), Xiaohang Yu <xhyumiracle@hyperoracle.io>, Huaizhe Xu (@HuaizheXu), Kartin <kartin@hyperoracle.io>
+discussions-to: https://ethereum-magicians.org/t/eip-7007-zkml-aigc-nfts-an-erc-721-extension-interface-for-zkml-based-aigc-nfts/14216
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-05-10
+requires: 165, 721
+---
+
+## Abstract
+
+The zkML AIGC-NFTs standard is an extension of the [ERC-721](./eip-721.md) token standard for AI Generated Content. It proposes a set of interfaces for basic interactions and enumerable interactions for AIGC-NFTs. The standard includes a new mint event and a JSON schema for AIGC-NFT metadata. Additionally, it incorporates zkML capabilities to enable verification of AIGC-NFT ownership. In this standard, the `tokenId` is indexed by the `prompt`.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+**Every compliant contract must implement the [ERC-7007](./eip-7007.md), [ERC-721](./eip-721.md), and [ERC-165](./eip-165.md) interfaces.**
+
+The zkML AIGC-NFTs standard includes the following interfaces: 
+
+ERC-7007: Defines a mint event and a mint function for minting AIGC-NFTs. It also includes a verify function to check the validity of a combination of prompt and proof using zkML techniques.
+
+```solidity
+pragma solidity ^0.8.18;
+
+/**
+ * @dev Required interface of an ERC7007 compliant contract.
+ * Note: the ERC-165 identifier for this interface is 0x7e52e423.
+ */
+interface IERC7007 is IERC165, IERC721 {
+    /**
+     * @dev Emitted when `tokenId` token is minted.
+     */
+    event Mint(
+        uint256 indexed tokenId,
+        bytes indexed prompt,
+        bytes indexed aigcData,
+        string uri,
+        bytes proof
+    );
+
+    /**
+     * @dev Mint token at `tokenId` given `prompt`, `aigcData`, `uri` and `proof`.
+     *
+     * Requirements:
+     * - `tokenId` must not exist.'
+     * - verify(`prompt`, `aigcData`, `proof`) must return true.
+     *
+     * Optional:
+     * - `proof` should not include `aigcData` to save gas.
+     */
+    function mint(
+        bytes calldata prompt,
+        bytes calldata aigcData,
+        string calldata uri,
+        bytes calldata proof
+    ) external returns (uint256 tokenId);
```

Comment:
Generally minting is an application-specific operation. In other words, it is unusual to standardize `mint` because how it works is usually tied pretty closely to each token.

The question I'll need you to answer is: will anyone other than the deployer of the contract (or the deployer's specific dapp, like OpenSea) be calling the mint function?

If not, you don't really need to standardize it, though I think adding the ML metadata extensions make sense.

Reply:
@SamWilsn This seems a peer review rather than an editoral review. Maybe put it in the FEM?